Ensure that job log is cleared when bulk execution completes with an error (#17977)
crate/crate: docs/appendices/release-notes/5.10.8.rst

- Fixed an issue that caused queries to not complete and return if a 
  ``CircuitBreakerException`` was thrown while constructing the result set.

> - Fixed an issue that caused a bulk insert interrupted by ``KILL`` or
>   a node restart to not be cleared from the :ref:`sys.jobs <sys-jobs>` table.

Reviewer: Isn't this also the case when some error happens during executing? And isn't this also related to update + delete?
```suggestion
- Fixed an issue that caused an interrupted bulk operation to not be cleared
  from the :ref:`sys.jobs <sys-jobs>` table.
```
